Build fails on Mac Mojave

Seems like gapid is using some deprecated enums and classes.
I tried to pass -Wno-deprecated-declarations compiler flag to bazel,
but I don't know how to exactly do it, because bazel detects the Xcode compiler and uses it.
Any ideas?

exec env - \
    APPLE_SDK_PLATFORM='' \
    APPLE_SDK_VERSION_OVERRIDE='' \
    PATH=/anaconda3/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Users/edward.seo/Library/Android/sdk/ndk-bundle:/Users/edward.seo/Library/Android/sdk/platform-tools:/usr/local/opt/qt/bin \
    XCODE_VERSION_OVERRIDE=10.1.0 \
  external/local_config_cc/wrapped_clang '-D_FORTIFY_SOURCE=1' -fstack-protector -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections -fdata-sections '-std=c++11' -iquote . -iquote bazel-out/host/genfiles -iquote bazel-out/host/bin -iquote external/com_google_protobuf -iquote bazel-out/host/genfiles/external/com_google_protobuf -iquote bazel-out/host/bin/external/com_google_protobuf -iquote external/bazel_tools -iquote bazel-out/host/genfiles/external/bazel_tools -iquote bazel-out/host/bin/external/bazel_tools -isystem external/com_google_protobuf/src -isystem bazel-out/host/genfiles/external/com_google_protobuf/src -isystem bazel-out/host/bin/external/com_google_protobuf/src -MD -MF bazel-out/host/bin/core/os/device/deviceinfo/cc/_objs/darwin_query/query.mm.d '-frandom-seed=bazel-out/host/bin/core/os/device/deviceinfo/cc/_objs/darwin_query/query.mm.o' -isysroot __BAZEL_XCODE_SDKROOT__ -g0 -g0 -x objective-c++ '-DGAPID_VERSION_AND_BUILD="1.2.1:"' -Werror -DTARGET_OS_OSX -no-canonical-prefixes -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -c bazel-out/host/bin/core/os/device/deviceinfo/cc/darwin_query/osx/query.mm.cc -o bazel-out/host/bin/core/os/device/deviceinfo/cc/_objs/darwin_query/query.mm.o)
Execution platform: @bazel_tools//platforms:host_platform

Use --sandbox_debug to see verbose messages from the sandbox
bazel-out/host/bin/core/os/device/deviceinfo/cc/darwin_query/osx/query.mm.cc:35:3: error: 'NSOpenGLPixelFormat' is deprecated: first deprecated in macOS 10.14 - Please use Metal or MetalKit. [-Werror,-Wdeprecated-declarations]
  NSOpenGLPixelFormat* mGlFmt;
  ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSOpenGL.h:113:12: note: 'NSOpenGLPixelFormat' has been explicitly marked deprecated here
@interface NSOpenGLPixelFormat : NSObject <NSCoding>
           ^
bazel-out/host/bin/core/os/device/deviceinfo/cc/darwin_query/osx/query.mm.cc:36:3: error: 'NSOpenGLContext' is deprecated: first deprecated in macOS 10.14 - Please use Metal or MetalKit. [-Werror,-Wdeprecated-declarations]
  NSOpenGLContext* mGlCtx;
  ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSOpenGL.h:202:12: note: 'NSOpenGLContext' has been explicitly marked deprecated here
@interface NSOpenGLContext : NSObject <NSLocking>
           ^
bazel-out/host/bin/core/os/device/deviceinfo/cc/darwin_query/osx/query.mm.cc:65:3: error: 'NSOpenGLPixelFormatAttribute' is deprecated: first deprecated in macOS 10.14 [-Werror,-Wdeprecated-declarations]

.....

--- a/core/os/device/deviceinfo/cc/BUILD.bazel
+++ b/core/os/device/deviceinfo/cc/BUILD.bazel
@@ -17,7 +17,10 @@ load("//tools/build:rules.bzl", "android_dynamic_library", "cc_copts", "mm_libra
 mm_library(
     name = "darwin_query",
     srcs = glob(["osx/*.mm"]),
-    copts = cc_copts(),
+    copts = cc_copts() + [
+        # Avoid OpenGL deprecation error.
+        "-DGL_SILENCE_DEPRECATION",
+    ],
     copy_hdrs = ["query.h"],
     deps = ["//core/os/device:device_cc_proto"],
 )

--- a/gapir/cc/BUILD.bazel
+++ b/gapir/cc/BUILD.bazel
@@ -57,7 +57,12 @@ mm_library(
         ":gles_h",
         ":vulkan_h",
     ],
-    copts = cc_copts(),
+    copts = cc_copts() + [
+        # Avoid OpenGL deprecation error.
+        "-DGL_SILENCE_DEPRECATION",
+        # Avoid NSOpenGLContext.setView deprecation error.
+        "-Wno-deprecated-declarations",
+    ],
     deps = [
         "//core/cc",
     ],

